pkg/network/bridge/plugin.go

```diff
@@ -205,30 +206,94 @@ func (nbdp *NetworkBridgeDevicePlugin) attachPods() {
 func attachPodToBridge(bridgeName, nicName string, containerPid int) error {
 	linkName := randInterfaceName()
 
+	// fetch the bridge, this is expected to succeed since attachPodToBridge is invoked only if bridge exists
 	bridge, err := netlink.LinkByName(bridgeName)
 	if err != nil {
 		return err
 	}
 
-	link := &netlink.Veth{LinkAttrs: netlink.LinkAttrs{Name: linkName, MasterIndex: bridge.Attrs().Index}, PeerName: nicName}
+	// create the virtual interface that should be connected to the bridge
+	// TODO: allow setting MAC address for that interface
+	link := &netlink.Veth{
+		LinkAttrs: netlink.LinkAttrs{
+			Name: linkName,
+			MasterIndex: bridge.Attrs().Index,
+			MTU: bridge.Attrs().MTU, 
+			NetNsID: bridge.Attrs().NetNsID},
+		PeerName: nicName}
 
 	err = netlink.LinkAdd(link)
 	if err != nil {
 		return err
 	}
 
+	// set interface up
+	err = netlink.LinkSetUp(link)
+	if err!= nil {
```

Comment:
please run gofmt on the document

Reply:
strange, go fmt run as part of the makefile. anyway, fixed now.